classes/PrestaShopCollection.php

```diff
@@ -336,7 +336,7 @@ public function getAll($display_query = false)
         $this->is_hydrated = true;
 
         $alias = $this->generateAlias();
-        //$this->query->select($alias.'.*');
+        $this->query->select('*, a0.`' . $this->definition['primary'] . '`');
```

Comment:
Can you explain it. Why a0? where it comes from?

Reply:
PrestashopCollection class
  protected function generateAlias($association = '')
    {
        if (!isset($this->alias[$association])) {
            $this->alias[$association] = 'a'.$this->alias_iterator++;
        }
        return $this->alias[$association];
    }
When you call a multilingual collection, the first table gets alias a0

Reply:
Rather, any entity

Reply:
@PierreRambaud You can check? I read that this is a critical error

Reply:
This is why you must use `$alias` instead of `a0` :thinking:

Reply:
@PierreRambaud "a0" the first entity table always comes, otherwise you won’t do it, and without this addition, the collection to the prestashop becomes useless

Reply:
@PierreRambaud When you receive a collection of multilanguage entities, a choice is made of the type "a0. *, a1. *" If there is no data in the multilanguage, then "a1.id_table = null" overwrites "a0.id_table" and the collection is formed incorrectly

Reply:
@PierreRambaud See code please, where formed collection from data from database

Reply:
Can you give us more information about how to reproduce.
What do you mean by multilanguage entity and multilanguage table? 

As you can see the code if you have
```php
$alias = $this->generateAlias();
```
`$alias` => `a0`.

Otherwise you need to retrieve the real alias, not writing something like something which can be dynamic.
What if in the future we change the generateAlias and use `b.iterator` instead `a.iterator`. We will have a bug because the alias getter wasn't used :/

Reply:
@pierreavizou I updated file PrestaShopCollection.php, replaced a0 on $alias

Reply:
@PierreRambaud Aaaa, noooo. You lose other fields, from table_lang

Reply:
@PierreRambaud  In ObjectModel::hydrateCollection an object is going from fields, please look, you suggested a bad option, my worker, I already added in my project it works very well

Reply:
You miss the `.` in your commit: https://github.com/PrestaShop/PrestaShop/pull/14169/commits/addfd67e8bb7a1f60f6cca2b7d4dca7f1c86561b#diff-9d88e764dbaf9a1eabd6fd2246342223R339

It should be `$this->query->select($alias . '.*');`